ci-operator/step-registry/mco/conf/day2/destroy-custom-mcp/mco-conf-day2-destroy-custom-mcp-commands.sh

```diff
@@ -0,0 +1,100 @@
+    local CMD="$1"
+    echo "Running command: ${CMD}"
+    eval "${CMD}"
+}
+
+# waiting for mcp become a given status
+function wait_mcp(){
+    local mcp_name="${1}" expected_updated="$2" expected_machine_count="$3"
+    local wait_time=30
+    while (( wait_time > 0 )); do
+        sleep 10
+        out="$(oc get machineconfigpools ${mcp_name} --no-headers)"
+        echo $out
+        updated="$(echo "${out}" | awk '{print $3}')"
+        MACHINECOUNT="$(echo "${out}" | awk '{print $6}')"
+
+        if [[ "${updated}" == "${expected_updated}" && "${MACHINECOUNT}" == "${expected_machine_count}" ]]; then
+            echo "MCP ${mcp_name} is ready for next step, continue..." && return 0
+        fi
+        wait_time=$(( wait_time - 1 ))
+    done
+    run_command "oc get machineconfigpools"
+    run_command "oc get nodes"
+    echo "MCP ${mcp_name} is not ready for next step, break..." && exit 1
+}
+
+function remove_custom_mcp() {
+    local mcp_name="$1"
+    local mcp_node_label="${2},node-role.kubernetes.io/${mcp_name}="
+    local -a matched_nodes_array=()
+
+    echo "Removing customer worker node label from mcp ${mcp_name}"
+    read -a matched_nodes_array <<< "$(oc get nodes -l ${mcp_node_label} -ojsonpath="{.items[:].metadata.name}")"
+    echo "Matched nodes with label ${mcp_node_label}: ${matched_nodes_array[*]}"
+
+    if [[ ${#matched_nodes_array[@]} == 0 ]]; then
+        echo "No matched nodes!"
+        return 0
+    fi
+
+    local out WORKER_MACHINECOUNT EXPECTED_WORKER_MACHINECOUNT
+    echo "Worker mcp status before removing custom mcp label:"
+    run_command "oc get machineconfigpools worker"
+    out="$(oc get machineconfigpools worker --no-headers)"
+    WORKER_MACHINECOUNT="$(echo "${out}" | awk '{print $6}')"
+    printf "\n"
+
+    for node in "${matched_nodes_array[@]}"; do
+        run_command "oc label node ${node} node-role.kubernetes.io/${mcp_name}-"
+    done
+
+    printf "\n"
+    echo "Waiting for customer mcp pool becomes empty"
+    wait_mcp ${mcp_name} "True" "0"
+
+    echo "Worker mcp status after removing customer mcp label:"
+    EXPECTED_WORKER_MACHINECOUNT=$(( ${#matched_nodes_array[@]} + ${WORKER_MACHINECOUNT} ))
+    wait_mcp "worker" "True" "${EXPECTED_WORKER_MACHINECOUNT}"
+
+    printf "\n"
+    run_command "oc delete mcp ${mcp_name}"
+}
+
+if  [[ -z "$MCO_CONF_DAY2_CUSTOM_MCP" ]]; then
+  echo "This installation does not need to create any custom MachineConfigPool"
+  exit 0
+fi
+set_proxy
+custom_mcp_num=$(echo "${MCO_CONF_DAY2_CUSTOM_MCP}" | jq -r ". | length")
+last_mcp_index=$((custom_mcp_num - 1))
+for index in $(seq 0 ${last_mcp_index}); do
+    custom_mcp_name=$(echo "${MCO_CONF_DAY2_CUSTOM_MCP}" | jq -r ".[$index].mcp_name")
+    if [[ -z "$custom_mcp_name" ]]; then
+        echo "No mcp_name input"
+        exit 255
+    fi
+    custom_mcp_node_label=$(echo "${MCO_CONF_DAY2_CUSTOM_MCP}" | jq -r ".[$index].mcp_node_label")
+    if [[ -z "$custom_mcp_node_label" ]]; then
```

Comment:
`custom_mcp_node_label` is only used in `mco-conf-day2-create-custom-mcp`, for destroy step, we only need to know `custom_mcp_name`, right?

Reply:
I am using the label to find worker nodes, so want keep it. Yes we can find worker nodes by only `node-role.kubernetes.io/${mcp_name}=`, but I'd like a more precise filter here.

```
function remove_custom_mcp() {
    local mcp_name="$1"
    local mcp_node_label="${2},node-role.kubernetes.io/${mcp_name}="
```

Reply:
@JianLi-RH can you review my last comment?

Reply:
Which one? @jianlinliu

Reply:
Ignore it, I saw you already updated in your latest commit.